src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -7,59 +7,54 @@ import Errors from '../Errors/Errors';
 import Authors from '../Authors/Authors';
 import {
 	TEXT_ADD_AUTHOR,
+	TEXT_CANCEL,
 	TEXT_CREATE_AUTHOR,
 	TEXT_CREATE_COURSE,
 	TEXT_REMOVE_AUTHOR,
 } from '../../constants';
 
-import { pipeDuration } from '../../helpers/pipeDuration';
+import { DataContext } from '../../contexts/DataContext';
 
-import courseService from '../../services/course.service';
+import { pipeDuration } from '../../helpers/pipeDuration';
 
 import styles from './create-course.module.css';
 
 const CreateCourse = ({ handleBack }) => {
-	const { addCourse, addAuthor } = courseService;
+	const { createCourse, createAuthor } = useContext(DataContext);
+
 	const [errors, setErrors] = useState(null);
 	const [author, setAuthor] = useState('');
-	const [course, setCourse] = useState({
-		title: '',
-		description: '',
-		duration: null,
-		authors: [],
-	});
+	const [title, setTitle] = useState('');
+	const [description, setDescription] = useState('');
+	const [duration, setDuration] = useState(0);
+	const [authorIds, setAuthorIds] = useState([]);
 
 	const handleCreateAuthor = () => {
-		addAuthor({ name: author });
-		setAuthor('');
-	};
-
-	const handleChangeCourse = (e) => {
-		const target = e.target;
-		setCourse({
-			...course,
-			[target.name]:
-				target.type === 'number' ? Number(target.value) : target.value,
-		});
+		try {
+			createAuthor({ name: author });
+			setAuthor('');
+			setErrors(null);
+		} catch (errors) {
+			setErrors(errors);
+		}
```

Comment:
can you explain why you are using trycatch here?

Reply:
the validation function returns errors that were found when sending data. When an error returns, we record these errors in the state and the component is rerendered